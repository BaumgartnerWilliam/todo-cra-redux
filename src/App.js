import React from 'react';
import { Provider } from 'react-redux';

import store from './store';
import Main from './containers/Main/Main';

function App() {
  return (
    <div className="App">
      <Provider store={store}>
        <Main />
      </Provider>
    </div>
  );
}

export default App;
