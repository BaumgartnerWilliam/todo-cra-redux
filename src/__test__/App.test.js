import React from 'react';
import { render, wait, fireEvent } from '@testing-library/react';
import App from '../App';

describe('app', () => {
  it('renders without errors', () => {
    const { container } = render(<App />);
    expect(container).toBeInTheDocument();
  });

  it('starts the app with 3 notes', () => {
    const { queryAllByLabelText } = render(<App />);
    expect(queryAllByLabelText('todo').length).toBe(3);
  });

  it('has an add button', () => {
    const { getByText } = render(<App />);
    getByText(/^Add$/);
  });

  it('can add and remove todos', async () => {
    const { getByPlaceholderText, queryAllByLabelText, getByText } = render(
      <App />
    );

    await wait(() => {
      fireEvent.change(getByPlaceholderText(/my todo note/), {
        target: { value: 'my new todo' }
      });
    });

    await wait(() => {
      fireEvent.click(getByText(/^Add$/));
    });

    const todos = queryAllByLabelText('todo');

    expect(todos.length).toBe(4);
    const deleteButton = todos[3].querySelector('[aria-label="delete-todo"]');

    await wait(() => {
      fireEvent.click(deleteButton);
    });

    expect(queryAllByLabelText('todo').length).toBe(3);
  });

  it('matches snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
