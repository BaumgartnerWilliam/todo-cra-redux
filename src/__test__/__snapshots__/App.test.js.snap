// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="mt-5 col-6 container"
    >
      <div
        class="col"
      >
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
            >
              Add Todo
            </label>
            <input
              aria-label="add-todo-input"
              class="form-control"
              name="addTodo"
              placeholder="my todo note"
              value=""
            />
          </div>
          <div>
            <button
              aria-label="submit-todo"
              class="btn btn-primary"
              disabled=""
              type="submit"
            >
              Add
            </button>
          </div>
        </form>
      </div>
      <div
        class="mt-5 container"
      >
        <ul
          aria-label="todo-list"
          class="list-group"
        >
          <li
            aria-label="todo"
            class="noselect list-group-item list-group-item-action d-flex align-items-center justify-content-between"
            tabindex="0"
          >
            <span
              aria-label="unchecked"
              style="padding-right: 16px;"
            />
            <span>
              note 1
            </span>
            <span
              aria-label="delete-todo"
              tabindex="0"
            >
              <svg
                class="bi bi-trash-fill"
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5a.5.5 0 0 0-1 0v7a.5.5 0 0 0 1 0v-7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </li>
          <li
            aria-label="todo"
            class="noselect list-group-item list-group-item-action d-flex align-items-center justify-content-between"
            tabindex="0"
          >
            <span
              aria-label="unchecked"
              style="padding-right: 16px;"
            />
            <span>
              note 2
            </span>
            <span
              aria-label="delete-todo"
              tabindex="0"
            >
              <svg
                class="bi bi-trash-fill"
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5a.5.5 0 0 0-1 0v7a.5.5 0 0 0 1 0v-7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </li>
          <li
            aria-label="todo"
            class="noselect list-group-item list-group-item-action d-flex align-items-center justify-content-between list-group-item-warning"
            tabindex="0"
          >
            <span
              aria-label="checked"
            >
              <svg
                class="bi bi-check2"
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span>
              note 3
            </span>
            <span
              aria-label="delete-todo"
              tabindex="0"
            >
              <svg
                class="bi bi-trash-fill"
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5a.5.5 0 0 0-1 0v7a.5.5 0 0 0 1 0v-7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
